docs/DeveloperGuide.md

```diff
@@ -302,6 +303,35 @@ in a collection of property T matches any of the keywords.
 The use of generic types in the above predicates allows it to be implemented independent of the actual type
 of the property, as long as the relevant getters are supplied.
 
+
+### Feature: Substitution
+
+#### Overview
+
+The `sub` command allows the user to search for commonly used substitutions for ingredients.
+The following sequence of diagram illustrates how the different components interact with each other
+in the execution of a `sub soy sauce` command.
+
+<img src ="images/SubSequenceDiagram.png" width="1128" />
```

Comment:
personally i think the sequence diagram doesn't really add value cause it's executed like a regular command

i included a sequence diagram for `find` because i wanted to showcase how the predicates would be different for different properties, but i think `sub` doesn't do anything thats really different from a normal command

Reply:
true, i think i will remove it and just add a comment that it follows the regular command pipeline